apps/web/src/stores/documentVersions.ts

```diff
@@ -0,0 +1,118 @@
+  { currentDocument }: { currentDocument: SidebarDocument | undefined },
+  opts?: SWRConfiguration,
+) {
+  const { toast } = useToast()
+  const router = useRouter()
+  const { project } = useCurrentProject()
+  const { commit } = useCurrentCommit()
+  const { mutate, data, ...rest } = useSWR<DocumentVersion[]>(
+    ['documentVersions', project.id, commit.id],
+    async () => {
+      const [fetchedDocuments, errorFetchingDocuments] =
+        await getDocumentsAtCommitAction({
+          projectId: project.id,
+          commitId: commit.id,
+        })
+
+      if (errorFetchingDocuments) {
+        toast({
+          title: 'Creating file failed',
+          description:
+            errorFetchingDocuments.formErrors?.[0] ||
+            errorFetchingDocuments.message,
+          variant: 'destructive',
+        })
+        return []
+      }
+
+      return fetchedDocuments
+    },
+    opts,
+  )
+  const { execute: executeCreateDocument } = useServerAction(
+    createDocumentVersionAction,
+  )
+  const { execute: executeDestroyDocument } = useServerAction(
+    destroyDocumentAction,
+  )
+  const createFile = useCallback(
+    async ({ path }: { path: string }) => {
+      const [document, error] = await executeCreateDocument({
+        path,
+        projectId: project.id,
+        commitUuid: commit.uuid,
+      })
+
+      if (error) {
+        toast({
+          title: 'Creating document failed',
+          description: error.formErrors?.[0] || error.message,
+          variant: 'destructive',
+        })
+      } else if (document) {
+        const prevDocuments = data || []
+
+        if (document) {
+          mutate([...prevDocuments, document])
+          router.push(
+            ROUTES.projects
+              .detail({ id: project.id })
+              .commits.detail({ uuid: commit.uuid })
+              .documents.detail({ uuid: document.documentUuid }).root,
+          )
+        }
+      }
+    },
+    [executeCreateDocument, mutate, data],
+  )
+
+  const destroyFile = useCallback(
+    async (documentUuid: string) => {
+      const [_, error] = await executeDestroyDocument({
+        documentUuid,
+        projectId: project.id,
+        commitId: commit.id,
+      })
+      if (error) {
+        toast({
+          title: 'Deleting document failed',
+          description: error.formErrors?.[0] || error.message,
+          variant: 'destructive',
+        })
+      } else {
+        const prevDocuments = data || []
+        mutate(prevDocuments.filter((d) => d.documentUuid !== documentUuid))
+        if (currentDocument?.documentUuid === documentUuid) {
+          router.push(
+            ROUTES.projects
+              .detail({ id: project.id })
+              .commits.detail({ uuid: commit.uuid }).documents.root,
+          )
+        }
+      }
+    },
+    [executeDestroyDocument, mutate, data],
+  )
+
+  return { ...rest, documents: data ?? [], createFile, destroyFile, mutate }
```

Comment:
minor nitpicks: I'd prefer to keep it consistent between stores and always return data instead of a different-named prop that contains the data for each store. Also no need for the suffix "file" in actions. create and destroy is enough and beautifully concise.

Reply:
We'll have `createFolder` in this store